src/core/simulation_results.jl

```diff
@@ -1,517 +1,378 @@
-
-# Arguments
-- `file_path::String`: the file path to the dated folder with the raw results
-- `stage_number::String`: The stage of the results getting parsed
-
-# Example
-```julia
-execute!(simulation)
-results = load_simulation_results("file_path", "stage_name")
-```
-# Accepted Key Words
-"""
-function load_simulation_results(path::String, stage_name::String, kwargs...)
-    sim_results = deserialize_sim_output(path)
-    load_simulation_results(sim_results, stage_name; kwargs...)
-end
-"""
-    load_simulation_results(sim_output, stage)
-
-This function goes through the reference table of file paths and
-aggregates the results over time into a struct of type OperationsProblemResults
-
-# Arguments
-- `sim_output::SimulationResultsReference`: the container for the reference dictionary created in execute!
-- `stage_number::String`: The stage of the results getting parsed
+    Returns the values for the requested variable names. It keeps requests when performing multiple retrievals. Accepts a vector of names for the return of the values
 
-# Example
-```julia
-sim_output = execute!(simulation)
-results = load_simulation_results(sim_output, "stage_name")
-```
-# Accepted Key Words
+    # Accepted Key Words
+    - `initial_time::Dates.DateTime` : initial of the requested results
+    - `count::Int`: Number of results
 """
-function load_simulation_results(
-    sim_output::SimulationResultsReference,
-    stage_name::String;
-    kwargs...,
+function get_variables_values!(
+    res::SimulationResults,
+    names::Vector{Symbol};
+    initial_time::Union{Nothing, Dates.DateTime} = nothing,
+    count::Union{Int, Nothing} = nothing,
 )
-    results_folder = mkpath(joinpath(sim_output.results_folder, stage_name))
-    stage = "stage-$stage_name"
-    references = sim_output.ref
-    base_power = sim_output.base_powers[stage_name]
-    variables = Dict{Symbol, DataFrames.DataFrame}()
-    duals = Dict{Symbol, Any}()
-    params = Dict{Symbol, DataFrames.DataFrame}()
-    variable = (collect(keys(references[stage])))
-    dual = _find_duals(variable)
-    param = _find_params(collect(keys(references[stage])))
-    variable = setdiff(variable, vcat(param, dual))
-    time_stamp = DataFrames.DataFrame(Range = Dates.DateTime[])
-    time_length = sim_output.chronologies[stage]
-
-    for l in 1:length(variable)
-        date_df = references[stage][variable[l]]
-        variables[(variable[l])] = DataFrames.DataFrame()
-        for (ix, time) in enumerate(date_df.Date)
-            file_path = date_df[ix, :File_Path]
-            var = read_arrow_file(file_path)
-            variables[(variable[l])] = vcat(variables[(variable[l])], var[1:time_length, :])
-            if l == 1
-                time_stamp = vcat(time_stamp, _read_time(file_path, time_length))
-                check_file_integrity(dirname(file_path))
-            end
-        end
+    if initial_time === nothing
```

Comment:
The logic in this function appears identical to the logic for duals and parameters. You could eliminate the redundancy with a helper function (and keep these explicitly-named functions).